docs/samples/Microsoft.ML.AutoML.Samples/RecommendationExperiment.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.AutoML.Samples.DataStructures;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.AutoML.Samples
+{
+    public static class RecommendationExperiment
+    {
+        private static string TrainDataPath = "<Path to your train dataset goes here>";
+        private static string TestDataPath = "<Path to your test dataset goes here>";
+        private static string ModelPath = @"<Desired model output directory goes here>\Model.zip";
+        private static string LabelColumnName = "Rating";
+        private static uint ExperimentTime = 60;
+
+        public static void Run()
+        {
+            MLContext mlContext = new MLContext();
+
+            // STEP 1: Load data
+            IDataView trainDataView = mlContext.Data.LoadFromTextFile<Movie>(TrainDataPath, hasHeader: true, separatorChar: ',');
+            IDataView testDataView = mlContext.Data.LoadFromTextFile<Movie>(TestDataPath, hasHeader: true, separatorChar: ',');
+
+            var settings = new RecommendationExperimentSettings(RecommendationExperimentScenario.MatrixFactorization, "UserId", "MovieId")
+            {
+                MaxExperimentTimeInSeconds = ExperimentTime
+            };
+            var inputColumnInformation = new ColumnInformation();
+            inputColumnInformation.LabelCategoricalColumnNames.Add("MovieId");
+            inputColumnInformation.LabelCategoricalColumnNames.Add("UserId");
+            inputColumnInformation.LabelColumnName = "Rating";
+
+            // STEP 2: Run AutoML experiment
+            Console.WriteLine($"Running AutoML recommendation experiment for {ExperimentTime} seconds...");
+            ExperimentResult<RegressionMetrics> experimentResult = mlContext.Auto()
+                .CreateRecommendationExperiment(settings)
+                .Execute(trainDataView, inputColumnInformation);
+
+            // STEP 3: Print metric from best model
+            RunDetail<RegressionMetrics> bestRun = experimentResult.BestRun;
```

Comment:
@justinormont @eerhardt 
Are we OK with using RegressionMetrics for Recommendation tasks? (assuming no)
Or should I create a new RecommendationMetrics task? (assuming yes)

Reply:
This comment answers the question for RegressionMetrics: https://github.com/dotnet/machinelearning/pull/4246/files#r328404787

Seems like it's OK for now, until there are more trainers added

Reply:
The `Recommendation` evaluation in ML.NET uses `RegressionMetrics`, so I think we are safe.

https://github.com/dotnet/machinelearning/blob/3f98485e887177f2082f027c22f46fd64ebbe246/src/Microsoft.ML.Recommender/RecommenderCatalog.cs#L101-L117